yaml/build.go

```diff
@@ -0,0 +1,101 @@
+package yaml
+
+import (
+	"fmt"
+	"strconv"
+)
+
+// Build represents a build element in compose file.
+// It can take multiple form in the compose file, hence this special type
+type Build struct {
+	Context    string
+	Dockerfile string
+	Args       map[string]string
+}
+
+// MarshalYAML implements the Marshaller interface.
+func (b Build) MarshalYAML() (tag string, value interface{}, err error) {
+	m := map[string]interface{}{}
+	if b.Context != "" {
+		m["context"] = b.Context
+	}
+	if b.Dockerfile != "" {
+		m["dockerfile"] = b.Dockerfile
+	}
+	if len(b.Args) > 0 {
+		m["args"] = b.Args
+	}
+	return "", m, nil
+}
+
+// UnmarshalYAML implements the Unmarshaller interface.
+func (b *Build) UnmarshalYAML(tag string, value interface{}) error {
+	switch v := value.(type) {
+	case string:
+		b.Context = v
+	case map[interface{}]interface{}:
+		for mapKey, mapValue := range v {
+			switch mapKey {
+			case "context":
+				b.Context = mapValue.(string)
+			case "dockerfile":
+				b.Dockerfile = mapValue.(string)
+			case "args":
+				args, err := handleBuildArgs(mapValue)
+				if err != nil {
+					return err
+				}
+				b.Args = args
+			default:
+				// Ignore unknown keys
+				continue
+			}
+		}
+	default:
+		return fmt.Errorf("Failed to unmarshal Build: %#v", value)
+	}
+	return nil
+}
+
+func handleBuildArgs(value interface{}) (map[string]string, error) {
```

Comment:
Maybe we shouldn't make these two functions specific to build arguments. We'll probably have to do something similar for logging arguments, so maybe we can move these to a `utils.go`?